x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SpatialRelatesQuery.java

```diff
@@ -139,12 +141,13 @@ public String queryName() {
 
         @Override
         public float boost() {
-            return 0;
+            return boost;
         }
 
         @Override
         public QueryBuilder boost(float boost) {
-            throw new UnsupportedOperationException("Unimplemented: float");
+            this.boost = boost;
```

Comment:
do we ever expect this to be called with a value different from `0`? should we add an assertion if we always expect the boost to be 0?

Reply:
Right now there's no other expectation, but I think it's better to honor the `QueryBuilder` interface correctly and check that the boosting is done as expected on the tests.